src/Lecture1.hs

```diff
@@ -119,4 +127,15 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater :: Int -> [Int] -> String
+lowerAndGreater n list = show n ++ " is greater than " ++ (show . getLower 0) list ++ " elements and lower than " ++ (show . getGreater 0) list ++ " elements"
```

Comment:
That's correct! 💯 

However, with this solution, you will walk through the list twice.
You can write it more performant with only one traversal of the list 🙂
It's desirable to write functions that traverse lists single time and consume elements only once. In that case, the composition of such functions will have nice streaming and lazy properties.

Reply:
thanks! And do you use transducers in Haskell at all? The case is I work with ClojureScript so love them a lot 😄 And the fact that most of functions for working with collections are transducers is a nice bonus)

Reply:
I'm not familiar with transducers in ClojureScript 🤔 
Due to lazy evaluation, every function that consumes lists in Haskell only once is automatically streaming and consumes constant memory if written properly 🙂

Reply:
Thanks a lot for your comments! I reimplemented the solution. Will resolve this for now to not waste your time on re-check